safehttp/fileserver.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package safehttp
+
+import (
+	"errors"
+	"net/http"
+)
+
+// FileServer returns a handler that serves HTTP requests with the contents of
+// the file system rooted at root.
+func FileServer(root string) Handler {
+	fileServer := http.FileServer(http.Dir(root))
+
+	return HandlerFunc(func(rw ResponseWriter, req *IncomingRequest) Result {
+		fileServerRW := &fileServerResponseWriter{flight: rw.(*flight), header: http.Header{}}
+		fileServer.ServeHTTP(fileServerRW, req.req)
+		return fileServerRW.result
+	})
+}
+
+type fileServerResponseWriter struct {
+	flight *flight
```

Comment:
does that mean response in-flight? Is this some broader known convention in Go?

Reply:
It's the part of the framework that needs to be run per response in-flight, yes. It's not a broadly known convention in Go.